Homework11: Ekaterina Zavadskaya

KateZavadskaya opened the pull request:
Add done homework11

Docs: hm11_1.py, hm11_2.py

KateZavadskaya committed:
Feat(hm11): add hm11

Add draft hm11_1.py, add empty 11_2.py

Docs: hm11_1.py, hm11_2.py

KateZavadskaya committed:
Feat(doc): add code to files

Docs: hm11_1.py, hm11_2.py

KateZavadskaya committed:
Feat(hm11): done hm11

hm11_1.py code works, but without users
hm11_2.py code works

Docs: hm11_1.py, hm11_2.py

Stml89 commented on hm11_1.py:
```diff
@@ -0,0 +1,105 @@
+
+class Book:
+
+    """ h """
```
Please write good docstring

KateZavadskaya replied:
Done

Stml89 commented on hm11_1.py:
```diff
@@ -0,0 +1,105 @@
+
+        return f"{self.book_name} Врезерве:{self.rsrved}, наруках:{self.taken}"
+
+    # def take_book_status(self):
```
remove ti

KateZavadskaya replied:
Made a minor fix, but leave it in the code - think it should be a part of general logit and code.

Stml89 replied:
The goal is to have a clear codebase and keep only what you are using, so use it or remove

KateZavadskaya replied:
@Stml89 , add persent to __init__(11_2), also delete extra code (11_1).
Pylint / build (pull_request) is faild, yes - "R0913: Too many arguments (8/5) (too-many-arguments)". Have already disscissed with you this in Discord, please. 
Plaese, check my push despite Pylint🙏

Stml89 commented on hm11_1.py:
```diff
@@ -0,0 +1,105 @@
+book_user_2 = BookUser("Дима")
+
+
+book_1 = Book()
```
pass book information here, and do not hardcode it in the class

KateZavadskaya replied:
But it hardcode tatol result, as for me. I think it is easier once write it in code, then write in result.
I understand your point!) Will also do my best about this point in next HMs.

Stml89 replied:
Please, fix it here(this HW)

KateZavadskaya replied:
Just tried to explain my point of view. And may be was not correct in my message 🙏
Got you!) Will fix! 👍

Stml89 commented on hm11_1.py:
```diff
@@ -0,0 +1,105 @@
+
+        """ h """
+
+        if self.taken is False:
```
`if not self.taken:`

KateZavadskaya replied:
Done

Stml89 commented on hm11_2.py:
```diff
@@ -0,0 +1,36 @@
+        self.dep_time = dep_time * 12
+        self.dep_n = dep_n
+
+    def dep(self, persent):
```
typo `persent`

KateZavadskaya replied:
Done

Stml89 commented on hm11_2.py:
```diff
@@ -0,0 +1,36 @@
+class Bank:
+
+    """ h """
+    persent = 30
```
where are you using this class attribute?

KateZavadskaya replied:
It is the Pylinter wishes :(
hm11_2.py:17:8: W0201: Attribute 'percent' defined outside init (attribute-defined-outside-init)

Stml89 replied:
why you can not move it to `__init__`?
```
    def __init__(self, summ_dep, dep_time, dep_n, persent=0.1):
        self.summ_dep = summ_dep
        self.dep_time = dep_time * 12
        self.dep_n = dep_n
        self.persent = persent
```

Stml89 commented on hm11_2.py:
```diff
@@ -0,0 +1,36 @@
+        self.persent = persent
+        monthly_persent = persent/100/12
+        persent_total = self.summ_dep * (1 + monthly_persent) ** self.dep_time
+        return persent_total
```
round final amount to 2 digits after `.`

KateZavadskaya replied:
Done

Stml89 commented on hm11_1.py:
```diff
@@ -0,0 +1,105 @@
+
+
+book_user_1 = BookUser("Катя")
+book_user_2 = BookUser("Дима")
```
how are you using this 2 users?

In general, user come to the library and trying to do something with books:
``
book_user_1.reserve_book(book1)
book_user_2.reserve_book(book2)

book_user_1.take_book(book1)
book_user_2.take_book(book2)
``

KateZavadskaya replied:
Make some changes, now it seems to library user.

KateZavadskaya committed:
Fix(code): fix comments to code

hm11_1.py add 'user' to logic, write correct docstring, delete extra #
hm11_2.py correct typo, round final amount to 2 digits

Docs: hm11_1.py, hm11_2.py

KateZavadskaya committed:
Fix(code): fix comments to code

hm11_1.py delete extra code
hm11_2.py 'persent' move to __init__

Docs: hm11_1. py, hm11_2.py
